Modules/SubModule_ASL/xASL_wrp_ResampleASL.m

```diff
@@ -222,18 +223,38 @@ function xASL_wrp_ResampleASL(x)
     elseif round(dim4/2)~=dim4/2
         warning('Odd number of control-label pairs, skipping');
         return;
-    else    
+    else
+        
         % Paired subtraction
         [ControlIm, LabelIm] = xASL_quant_GetControlLabelOrder(ASL_im);
         ASL_im = ControlIm - LabelIm;
-        % Save PWI4D
-        fprintf('%s', [PathPWI4D{iSpace} ', ']);
-        xASL_io_SaveNifti(PathASL{iSpace}, PathPWI4D{iSpace}, ASL_im, 32, false);
+        % preallocate PWI
+        PWI = zeros(size(ASL_im,1),size(ASL_im,2),size(ASL_im,3));
+        
         % Average PWI
-        PWI = xASL_stat_MeanNan(ASL_im, 4);
-        % Save PWI
-        fprintf('%s\n', PathPWI{iSpace});
-        xASL_io_SaveNifti(PathASL{iSpace}, PathPWI{iSpace}, PWI, 32, false);
+        if isfield(x.modules.asl,'bMultiPLD') && x.modules.asl.bMultiPLD 
+            % multiPLD PWI
+            RepeatsInPLD = size(ASL_im,4)/length(x.Q.Initial_PLD); % we still need to create a single Repeats in X struct
+            for PLDnmbr = 1:length(x.Q.Initial_PLD)
+                RepeatsLocStartPerPLD = 1 + (PLDnmbr-1) * RepeatsInPLD;
+                RepeatsLocEndPerPLD = PLDnmbr * RepeatsInPLD;
+                PWI(:,:,:,PLDnmbr) = xASL_stat_MeanNan(ASL_im(:,:,:,RepeatsLocStartPerPLD:RepeatsLocEndPerPLD), 4);
+            end
+            % Save PWI
+            fprintf('%s\n', PathPWI4D{iSpace});
+            xASL_io_SaveNifti(PathASL{iSpace}, PathPWI4D{iSpace}, PWI, 32, false);
+            
+            PWIsingle = xASL_stat_MeanNan(ASL_im, 4); % create single PWI for further steps in ASL module
+            % Save single PWI
+            fprintf('%s\n', PathPWI{iSpace});
+            xASL_io_SaveNifti(PathASL{iSpace}, PathPWI{iSpace}, PWIsingle, 32, false);

```

Comment:
Nice work! What does iSpace stand for? Should it be iDim (dimension)? Or iPLD? Just avoid confusion

Reply:
iSpace is either native or common space, this part runs the analysis for both if inputted